sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentServiceRequest.java

```diff
@@ -276,6 +280,15 @@ public void setByteBuffer(ByteBuffer byteBuffer) {
         this.contentAsByteArray = toByteArray(byteBuffer);
     }
 
+    public RetryWithException getLastRetryWithException() {
+        return this.lastRetryWithException;
+    }
+
+    public void setLastRetryWithException(RetryWithException exception) {
```

Comment:
GoneAndRetryWithRetryPolicy is created per operation. Why can't we remember the lastRetryException in the RetryPolicy itself?

Reply:
if it is possible I would move the result exception from request to the retryPolicy.